sw/host/opentitanlib/src/util/file.rs

```diff
@@ -2,22 +2,74 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::Result;
+use anyhow::{anyhow, ensure, Result};
 use nix::libc::c_int;
 use nix::poll;
+use pem_rfc7468::{Decoder, Encoder, LineEnding};
+
 use std::convert::TryInto;
 use std::fs::File;
 use std::io::{self, Read, Write};
 use std::os::unix::io::{AsRawFd, RawFd};
 use std::path::Path;
 use std::time::Duration;
 
+/// Trait for data that can be written to and read from PEM files.
+pub trait PemSerilizable: ToWriter + FromReader {
+    /// The label for the PEM file.
+    ///
+    /// Appears around the base64 encoded data.
+    /// -----BEGIN MY_LABEL-----
+    /// ...
+    /// -----END MY_LABEL-----
+    fn label() -> &'static str;
+
+    /// Write to PEM file with label from `Self::label()`.
+    fn write_pem_file(&self, path: &Path) -> Result<()> {
+        const MAX_PEM_SIZE: usize = 4096;
+
+        let mut bytes = Vec::<u8>::new();
+        self.to_writer(&mut bytes)?;
+
+        let mut buf = [0u8; MAX_PEM_SIZE];
+        let mut encoder =
+            Encoder::new(Self::label(), LineEnding::LF, &mut buf).map_err(|e| anyhow!(e))?;
+        encoder.encode(&bytes).map_err(|e| anyhow!(e))?;
+        let len = encoder.finish().map_err(|e| anyhow!(e))?;
```

Comment:
It seems to be necessary

```
error[E0277]: the trait bound `pem_rfc7468::Error: StdError` is not satisfied
  --> sw/host/opentitanlib/src/util/file.rs:38:35
   |
38 |         let len = encoder.finish()?;
   |                                   ^ the trait `StdError` is not implemented for `pem_rfc7468::Error`
```

Reply:
Honestly I don't have the best understanding of error coercion. Could this be included as an additional error type in your `MyError` example below?

Something like this
```
#[derive(Debug, Error)]
enum PemError {
    #[error("PEM error: {0}")]
    InternalError(pem_rfc7468::Error),
    #[error("PEM type error; expecting {0} but got {1}")]
    LabelError(String, String),
}
```

And then
```
let len = encoder.finish().map_err(PemError::InternalError)?;
``` 

Still requires the `map_error()` call, so likely there is a better way.

Reply:
I didn't notice this earlier, but I think you need to enable the `std` feature on the PEM and RSA crates so that they'll `impl StdError` for their errors.